### ISSUE TITLE
Android HTTPClient lossy conversion

### ISSUE THREAD

**Issue by Informate** [tidev/titanium-sdk#14059]
### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

In the Ti.Network.HTTPClient the method send is not consistent between Android and iOS.
In Android the send method performs a data type conversion error on the POST request parameters, while on iOS the conversion is done correctly.
The long integer data type is wrongly converted to scentific formant, corrupting for example timestamps generated by Date.now().

FIXING WORKAROUND: Convert the long integer to String before passing it to the send method!

### Expected Behavior

Consistent behaviour between iOS and Android.
Non lossy data transfer and integrity of long integer data types.

### Actual behavior

Android wrongly converts long integer  to float scentific notation.

### Reproducible sample

HttpClientInstance.send( { timestamp: Date.now() } );

FIXING WORKAROUND: HttpClientInstance.send( { timestamp: Date.now().toString() } );

### Steps to reproduce

Crate an instance of HTTPClient, and make a request with a long integer in the parameters of the first argument of the send method, for example using Date.now().

### Platform

Android

### SDK version you are using

12.3.0.GA

**Comment by m1ga** [comment]
Not have any solution here but I've set a breakpoint in the send method and I can see that the parameter that is passed to that method is already a double:
![Bildschirmfoto_20240611_093807](https://github.com/tidev/titanium-sdk/assets/4334997/7ffb61cd-41b4-4143-894a-68192c775448)

**Comment by Informate** [comment]
> Not have any solution here but I've set a breakpoint in the send method and I can see that the parameter that is passed to that method is already a double: ![Bildschirmfoto_20240611_093807](https://private-user-images.githubusercontent.com/4334997/338483766-7ffb61cd-41b4-4143-894a-68192c775448.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwOTcyNDIsIm5iZiI6MTcxODA5Njk0MiwicGF0aCI6Ii80MzM0OTk3LzMzODQ4Mzc2Ni03ZmZiNjFjZC00MWI0LTQxNDMtODk0YS02ODE5MmM3NzU0NDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTFUMDkwOTAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzVjZGM4MzRmYWRiMDJjMWIwNGNiOTc4OGY2Zjg4ZGY4OTM3NTExODAzYjkyZWY1YzA3N2Q5Yjk4MmM5ZTUxNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.uDc6IWTucosGGeiqx8kEL7fWXejR6UMyEZPiNCfA-8g)

Thanks for your great support, maybe the problem occurs even then when the object attribute is set? Can you confirm that you initially see the number correctly, and that it is not the debugger doing the conversion incorrectly?

**Comment by m1ga** [comment]
yes, I'm setting `12345678901` as a value (or `Date.now()`).

**Comment by Informate** [comment]
> yes, I'm setting `12345678901` as a value (or `Date.now()`).

It seems to be Date.now() being cast to double instead of Long then.

**Comment by m1ga** [comment]
corrent. But it's all long numbers. With to String it will be a string so it doesn't need to convert it. But I don't know at which point the conversion is

**Comment by Informate** [comment]
> correct. But it's all long numbers. With to String it will be a string so it doesn't need to convert it. But I don't know at which point the conversion is

No, the Debugger is doing wrong!! :-/
It's not only send, but also the alert function called with Date.now() to rise the error!
I suppose that: If the toString conversion is done at JS level is ok, while if done at Java level is wrong!

**Comment by m1ga** [comment]
>  I suppose that: If the toString conversion is done at JS level is ok, while if done at Java level is wrong!

I don't think it is doing a `toString` in Java as it still shows it as a number (double as you can see in the debug screenshot). But still: not a long as it should be.

**Comment by Informate** [comment]
JS could be internally keeping the long integer inside a "magic" double for its triky casting all types in one place magic conversion, and under Android the number could be not correctly inverted to long.

**Comment by Informate** [comment]
I knew that JS try to fit all the datatypes in one putting pointers and other into the NaN space, but the long should not fit into a double NaN space.
